transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,251 @@
+            accessor=lambda otc_service: otc_service.get("seasonal_end"),
+        ),
+        CSVColumn(
+            header="Effective stale date due to end date",
+            accessor=lambda otc_service: otc_service.get(
+                "effective_stale_date_end_date"
+            ),
+        ),
+        CSVColumn(
+            header="Effective stale date due to Effective last modified date",
+            accessor=lambda otc_service: otc_service.get(
+                "effective_stale_date_last_modified_date"
+            ),
+        ),
+        CSVColumn(
+            header="Last modified date < Effective stale "
+            "date due to OTC effective date",
+            accessor=lambda otc_service: otc_service.get(
+                "last_modified_lt_effective_stale_date_otc"
+            ),
+        ),
+        CSVColumn(
+            header="Effective stale date due to OTC effective date",
+            accessor=lambda otc_service: otc_service.get(
+                "effective_stale_date_otc_effective_date"
+            ),
+        ),
+    ]
+
+    def __init__(self, organisation_id: int):
+        self._organisation_id = organisation_id
+        self._object_list = []
+
+    def _update_data(
+        self,
+        service: Optional[OTCService],
+        file_attribute: Optional[TXCFileAttributes],
+        seasonal_service: Optional[SeasonalService],
+        exemption: Optional[ServiceCodeExemption],
+        staleness_status: Optional[str],
+        require_attention: str,
+    ) -> None:
+        self._object_list.append(
+            {
+                "require_attention": require_attention,
+                "scope_status": exemption and exemption.registration_number,
+                "licence_number": service and service.otc_licence_number,
+                "service_code": service and service.registration_number,
+                "line_number": service and service.service_number,
+                "dataset_id": file_attribute and file_attribute.revision.dataset_id,
+                "seasonal_status": seasonal_service
+                and seasonal_service.seasonal_status,
+                "seasonal_start": seasonal_service and seasonal_service.start,
+                "seasonal_end": seasonal_service and seasonal_service.end,
+                "staleness_status": staleness_status,
+                "effective_seasonal_start_date": seasonal_service
+                and seasonal_service.start - timedelta(days=42),
+                "effective_stale_date_end_date": file_attribute
+                and file_attribute.effective_stale_date_end_date,
+                "effective_stale_date_last_modified_date": file_attribute
+                and file_attribute.effective_stale_date_last_modified_date,
+                "last_modified_lt_effective_stale_date_otc": service
+                and (
+                    service.last_modified.date()
+                    < service.effective_stale_date_otc_effective_date
+                ),
```

Comment:
This should be comparing the last modified date of the `TXCFileAttribute` object.